src/components/App/App.tsx

```diff
@@ -0,0 +1,45 @@
+import { FC, useEffect, useState } from "react";
+
+import Equation from "../../models/Equation";
+import Calculator from "../Calculator/Calculator";
+import WelcomeMessage from "../WelcomeMessage/WelcomeMessage";
+import EquationHistory from "../EquationHistory/EquationHistory";
+import styles from "./App.module.css";
+
+const timeForWelcome = 5000;
+
+const App: FC<{}> = (props) => {
+  const [history, setHistory] = useState<Equation[]>([]);
+  const [isWelcomeOver, setIsWelcomeOver] = useState<boolean>();
+
+  const addToHistory = (equtionToAdd: Equation) => {
+    if (equtionToAdd.isValidEquation()) {
+      const updatedHistory = [...history, equtionToAdd];
+      setHistory(updatedHistory);
+    }
+  };
+
+  useEffect(() => {
+    setTimeout(() => {
```

Comment:
when using timeout on useEffect you must to clear it when component die
`const timeout = setTimeout(() => {`
and then in the end: 
`() => clearTimeout(timeout)`